### PR TITLE
Fix steam boiler can accept steam

### PR DESCRIPTION
**Pull request by Phoupraw**

### CODE REVIEW

**Review by Phoupraw — COMMENTED** [review]
It makes no sense that #291 appears here. It's in 1.20 branch and should have been merged in #285.

**Comment by mikerooni** [comment]
Includes #291 - that one can be closed when this PR is merged.

File: common/src/main/java/com/gregtechceu/gtceu/api/machine/trait/NotifiableFluidTank.java [hunk 1]
```diff
@@ -224,7 +224,7 @@ public boolean isFluidValid(int tank, @NotNull FluidStack stack) {
 
     @Override
     public long fill(FluidStack resource, boolean simulate, boolean notifyChanges) {
-        if (resource.isEmpty()) return 0;
+        if (resource.isEmpty() || !getCapabilityIO().support(IO.IN)) return 0;
```
Comment (Yefancy) [inline common/src/main/java/com/gregtechceu/gtceu/api/machine/trait/NotifiableFluidTank.java]: use `canCapInput()` instead

**Review by Yefancy — DISMISSED** [review]

**Comment by Yefancy** [comment]
lgtm

**Review by mikerooni — APPROVED** [review]

**Comment by Phoupraw** [comment]
How can I resolve outdated requested change?

**Pull request merged**